### PR TITLE
Dịch "underfit-overfit_vn" - Phần 8

### LINKED ISSUE #885: Dịch "underfit-overfit_vn" - Phần 8
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_multilayer-perceptrons/underfit-overfit_vn.md#L456-L618

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by tranyenthy**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #885 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 1]
```diff
@@ -473,14 +473,15 @@ In part, the current success of deep learning owes to the current abundance of m
...
-## *dịch tiêu đề phía trên*
+## Hồi quy Đa thức 
 
 <!--
 We can now explore these concepts interactively by fitting polynomials to data.
 To get started we will import our usual packages.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ ta có thể khám phá một cách tương tác những khái niệm này bằng cách tìm các đa thức khớp với dữ liệu.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
Bây giờ ta có thể khám phá một cách tương tác những khái niệm này bằng cách khớp đa thức với dữ liệu.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 2]
```diff
@@ -473,14 +473,15 @@ In part, the current success of deep learning owes to the current abundance of m
...
+Để bắt đầu ta sẽ nhập các gói thông thường.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
Để bắt đầu ta sẽ nhập các gói thường dùng.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 3]
```diff
@@ -493,26 +494,28 @@ npx.set_np()
...
+### Tạo ra Tập dữ liệu
 
 <!--
 First we need data. Given $x$, we will use the following cubic polynomial to generate the labels on training and test data:
 -->
 
-*dịch đoạn phía trên*
+Đầu tiên ta cần dữ liệu. Cho $x$, ta sẽ sử dụng đa thức bậc ba ở dưới đây để tạo nhãn cho tập dữ liệu huấn luyện và tập kiểm tra:
 
 $$y = 5 + 1.2x - 3.4\frac{x^2}{2!} + 5.6 \frac{x^3}{3!} + \epsilon \text{ where }
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]: thiếu mất dấu comment ở đây
```suggestion
<!--
$$y = 5 + 1.2x - 3.4\frac{x^2}{2!} + 5.6 \frac{x^3}{3!} + \epsilon \text{ where }
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 4]
```diff
@@ -493,26 +494,28 @@ npx.set_np()
...
 \epsilon \sim \mathcal{N}(0, 0.1).$$
 -->
 
-*dịch đoạn phía trên*
+$$y = 5 + 1.2x - 3.4\frac{x^2}{2!} + 5.6 \frac{x^3}{3!} + \epsilon \text{ where }
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
$$y = 5 + 1.2x - 3.4\frac{x^2}{2!} + 5.6 \frac{x^3}{3!} + \epsilon \text{ với }
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 5]
```diff
@@ -493,26 +494,28 @@ npx.set_np()
...
+\epsilon \sim \mathcal{N}(0, 0.1).$$
 
 <!--
 The noise term $\epsilon$ obeys a normal distribution with a mean of 0 and a standard deviation of 0.1.
 We will synthesize 100 samples each for the training set and test set.
 -->
 
-*dịch đoạn phía trên*
+Thuật ngữ nhiễu $\epsilon$ tuân theo phân phối chuẩn (phân phối Gauss) với giá trị trung bình bằng 0 và độ lệch chuẩn bằng 0.1.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
Số hạng nhiễu $\epsilon$ tuân theo phân phối chuẩn (phân phối Gauss) với giá trị trung bình bằng 0 và độ lệch chuẩn bằng 0.1.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 6]
```diff
@@ -493,26 +494,28 @@ npx.set_np()
...
+Ta sẽ tổng hợp 100 mẫu cho mỗi tập huấn luyện và tập kiểm tra.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]: mình nghĩ từ `synthesize` cứ dịch là `tạo` cho đơn giản
```suggestion
Ta sẽ tạo 100 mẫu cho mỗi tập huấn luyện và tập kiểm tra.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 7]
```diff
@@ -537,14 +540,18 @@ Factorials are implemented in Gluon using the Gamma function,
 where $n! = \Gamma(n+1)$.
 -->
 
-*dịch đoạn phía trên*
+Để tối ưu hóa, ta thường muốn tránh các giá trị rất lớn của gradient, mất mát, v.v.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
Khi tối ưu hóa, ta thường muốn tránh các giá trị rất lớn của gradient, mất mát, v.v.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 8]
```diff
@@ -537,14 +540,18 @@ Factorials are implemented in Gluon using the Gamma function,
...
+Đây là lý do tại sao các đơn thức lưu trong `poly_features` được thay đổi kích thước từ $x^i$ thành $\frac{1}{i!} x^i$.
+Nó cho phép ta tránh các giá trị quá lớn với số mũ bậc cao $i$.
+Phép tính giai thừa được khai triển trong Gluon sử dụng hàm Gamma, khi $n! = \Gamma(n+1)$.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
Phép tính giai thừa được lập trình trong Gluon bằng hàm Gamma, với $n! = \Gamma(n+1)$.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 9]
```diff
@@ -537,14 +540,18 @@ Factorials are implemented in Gluon using the Gamma function,
...
 
 <!--
 Take a look at the first 2 samples from the generated dataset.
 The value 1 is technically a feature, namely the constant feature corresponding to the bias.
 -->
 
-*dịch đoạn phía trên*
+Hãy xem xét hai mẫu đầu tiên trong tập dữ liệu được tạo ra.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
Hãy xét hai mẫu đầu tiên trong tập dữ liệu được tạo.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 10]
```diff
@@ -554,13 +561,13 @@ features[:2], poly_features[:2], labels[:2]
...
 
-### *dịch tiêu đề phía trên*
+### Huấn luyện và Kiểm tra Mô hình
 
 <!--
 Let's first implement a function to evaluate the loss on a given data.
 -->
 
-*dịch đoạn phía trên*
+Trước tiên ta lập trình hàm để đánh giá mất mát của dữ liệu cho trước.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
Trước tiên ta lập trình hàm để tính giá trị mất mát của dữ liệu cho trước.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 11]
```diff
@@ -612,15 +619,17 @@ def train(train_features, test_features, train_labels, test_labels,
...
+### Khớp Hàm Đa thức Bậc Ba (dạng chuẩn)
 
 <!--
 We will begin by first using a third-order polynomial function with the same order as the data generation function.
 The results show that this model’s training error rate when using the testing dataset is low.
 The trained model parameters are also close to the true values $w = [5, 1.2, -3.4, 5.6]$.
 -->
 
-*dịch đoạn phía trên*
+Ta sẽ bắt đầu bằng cách sử dụng hàm đa thức bậc ba cùng bậc với hàm tạo dữ liệu.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
Ta sẽ bắt đầu với việc sử dụng hàm đa thức bậc ba, cùng bậc với hàm tạo dữ liệu.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 12]
```diff
@@ -612,15 +619,17 @@ def train(train_features, test_features, train_labels, test_labels,
...
+Kết quả cho thấy tỉ lệ lỗi huấn luyện thấp khi sử dụng tập kiểm tra trên mô hình này.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]: sao đã có `lỗi huấn luyện` lại còn có `tập kiểm tra` nhỉ :)))

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 13]
```diff
@@ -612,15 +619,17 @@ def train(train_features, test_features, train_labels, test_labels,
...
+Các tham số trên mô hình được huấn luyện cũng gần với giá trị thật $w = [5, 1.2, -3.4, 5.6]$.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]:
```suggestion
Các tham số của mô hình được huấn luyện cũng gần với giá trị thật $w = [5, 1.2, -3.4, 5.6]$.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md (hunk 12 above)
Comment (tranyenthy) [reply chapter_multilayer-perceptrons/underfit-overfit_vn.md]: Mình thấy mình dịch sát với từ quá không biết có bỏ qua ý nào của tác giả (bao hàm) không

Comment (minhduc0711) [reply chapter_multilayer-perceptrons/underfit-overfit_vn.md]: Chắc tác giả muốn nói cả lỗi huấn luyện lẫn lỗi kiểm tra đều thấp, nhưng viết lộn xộn.
Để đợi xem người khác có ý kiến gì không vậy.

Comment (minhduc0711) [reply chapter_multilayer-perceptrons/underfit-overfit_vn.md]: mình nghĩ nên linh động tự sửa bản gốc vậy

btw mình cũng bỏ từ `tỉ lệ`, quote anh Tiệp
> Uh, mình sẽ dịch là tỉ lệ lỗi nếu đang nói về các mô hình phân loại, ở đó tỉ lệ lỗi là tỉ lệ giữa các điểm bị phân loại sai trên tổng số điểm. Với bài toán hồi quy như bài này, tỉ lệ lỗi không có nghĩ vì không có mẫu số.

```suggestion
Kết quả cho thấy cả lỗi huấn luyện và lỗi kiểm tra của mô hình đều thấp.
```

**Comment by minhduc0711** [comment]
mình nghĩ nên linh động tự sửa bản gốc vậy

btw mình cũng bỏ từ `tỉ lệ`, quote anh Tiệp
> Uh, mình sẽ dịch là tỉ lệ lỗi nếu đang nói về các mô hình phân loại, ở đó tỉ lệ lỗi là tỉ lệ giữa các điểm bị phân loại sai trên tổng số điểm. Với bài toán hồi quy như bài này, tỉ lệ lỗi không có nghĩ vì không có mẫu số.

```suggestion
Kết quả cho thấy cả lỗi huấn luyện và lỗi kiểm tra của mô hình đều thấp.
```

File: chapter_multilayer-perceptrons/underfit-overfit_vn.md [hunk 14]
```diff
@@ -612,15 +620,17 @@ def train(train_features, test_features, train_labels, test_labels,
...
 
 <!--
 We will begin by first using a third-order polynomial function with the same order as the data generation function.
 The results show that this model’s training error rate when using the testing dataset is low.
 The trained model parameters are also close to the true values $w = [5, 1.2, -3.4, 5.6]$.
 -->
 
-*dịch đoạn phía trên*
+Ta sẽ bắt đầu với việc sử dụng hàm đa thức bậc ba, cùng bậc với hàm tạo dữ liệu.
+Kết quả cho thấy tỉ lệ lỗi huấn luyện thấp khi sử dụng tập kiểm tra trên mô hình này.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/underfit-overfit_vn.md]: mình nghĩ nên linh động tự sửa bản gốc vậy

btw mình cũng bỏ từ `tỉ lệ`, quote anh Tiệp
> Uh, mình sẽ dịch là tỉ lệ lỗi nếu đang nói về các mô hình phân loại, ở đó tỉ lệ lỗi là tỉ lệ giữa các điểm bị phân loại sai trên tổng số điểm. Với bài toán hồi quy như bài này, tỉ lệ lỗi không có nghĩ vì không có mẫu số.

```suggestion
Kết quả cho thấy cả lỗi huấn luyện và lỗi kiểm tra của mô hình đều thấp.
```
